include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -676,6 +676,18 @@ struct __result_and_scratch_storage : __result_and_scratch_storage_base
         }
     }
 
+    std::shared_ptr<__result_and_scratch_storage_base>
+    __clone_to_result_and_scratch_storage_base_ptr() const
```

Comment:
Why do we need such clone?
It was not before.

Reply:
This is some common implementation for creation of `std::shared_ptr<__result_and_scratch_storage_base>` pointer for hang-up `__result_and_scratch_storage` state on it.